File: backend/main_service/src/main/java/pl/michal_sobiech/engineering_thesis/api/TimeWindowMapper.java
package pl.michal_sobiech.engineering_thesis.api;

import java.time.LocalTime;

import org.SwaggerCodeGenExample.model.TimeWindow;

import pl.michal_sobiech.core.enterprise_service_slot_template.custom_appointments.CreateTimeWindowTemplateCommand;
import pl.michal_sobiech.core.enterprise_service_slot_template.custom_appointments.CustomSlotTemplate;

public class TimeWindowMapper {

    public static CreateTimeWindowTemplateCommand createCommandFromSwaggerTimeWindow(
            TimeWindow swaggerTimeWindow) {
        LocalTime startTime = LocalTime.parse(swaggerTimeWindow.getStartTime());
        LocalTime endTime = LocalTime.parse(swaggerTimeWindow.getEndTime());
        return new CreateTimeWindowTemplateCommand(
                DayOfWeekMapper.swaggerToStdDayOfWeek(swaggerTimeWindow.getDayOfWeek()),
                startTime,
                endTime);
    }

    public static TimeWindow fromCustomSlotTemplate(CustomSlotTemplate customSlotTemplate) {
        return new TimeWindow(
                DayOfWeekMapper.stdDayOfWeekToSwagger(customSlotTemplate.dayOfWeek()),
                customSlotTemplate.start().toString(),
                customSlotTemplate.end().toString());
    }

}

File: frontend/src/api/time-window-mapper.ts
import { LocalTime } from "@js-joda/core";
import { TimeWindow as SwaggerTimeWindow } from "../GENERATED-api";
import { WeeklyTimeWindow } from "../utils/WeeklyTimeWindow";
import { jodaDayOfWeekToSwagger, swaggerDayOfWeekToJoda } from "./day-of-week-mapper";

export function swaggerTimeWindowToWeeklyTimeWindow(swaggerTimeWindow: SwaggerTimeWindow): WeeklyTimeWindow {
    return {
        dayOfWeek: swaggerDayOfWeekToJoda(swaggerTimeWindow.dayOfWeek),
        start: LocalTime.parse(swaggerTimeWindow.startTime),
        end: LocalTime.parse(swaggerTimeWindow.endTime),
    };
}

export function weeklyTimeWindowToSwagger(domainTimeWindow: WeeklyTimeWindow): SwaggerTimeWindow {
    return {
        dayOfWeek: jodaDayOfWeekToSwagger(domainTimeWindow.dayOfWeek),
        startTime: domainTimeWindow.start.toString(),
        endTime: domainTimeWindow.end.toString()
    };
}